### PR TITLE
Some changes to group saving logic

### PR DESCRIPTION
**Pull request by hypeJunction**
**feat(groups): group ACL name is now always in sync with the group name**

Group name changes that take place outside of the groups/edit
action are now correctly reflected in access collections table

**feat(groups): group metadata ownership in now in sync with group ownership**

Group metadata ownerships is now correctly updated if group
owner changes outside of the groups/edit action

**feat(groups): validate container permissions when creating a new group**

Makes it easier to supply a container_guid for a new group without having to
replace an entire action.
Makes sure the user can write a new group to container, thus making it
easier to implement more granular controls over who can create new groups

### CODE REVIEW

File: mod/groups/start.php [hunk 1]
```diff
@@ -496,17 +496,50 @@ function groups_update_event_listener($event, $type, $group) {
...
+
+		// In addition to standard icons, groups plugin stores a copy of the original upload
+		$filehandler = new ElggFile();
+		$filehandler->owner_guid = $previous_owner_guid;
+		$filehandler->setFilename("groups/$group->guid.jpg");
+		$filehandler->transfer($group->owner_guid);
+
+		// Update owned metadata
+		$metadata = elgg_get_metadata([
+			'guid' => $group->guid,
```
Comment (hypeJunction) [inline mod/groups/start.php]: We could use `metadata_owner_guids` here, I suppose

File: mod/groups/start.php [hunk 2]
```diff
@@ -496,17 +496,50 @@ function groups_update_event_listener($event, $type, $group) {
...
-	if (empty($original_attributes['owner_guid'])) {
-		return;
-	}
 
-	$previous_owner_guid = $original_attributes['owner_guid'];
+	if (!empty($original_attributes['owner_guid'])) {
+		$previous_owner_guid = $original_attributes['owner_guid'];
+
+		// In addition to standard icons, groups plugin stores a copy of the original upload
+		$filehandler = new ElggFile();
```
Comment (hypeJunction) [inline mod/groups/start.php]: We don't need this anymore. Group icons are already owned by the group.

**Comment by mrclay** [comment]
So far so good

**Comment by hypeJunction** [comment]
I am done with this one.

**Comment by hypeJunction** [comment]
There was also an issue with group icon upgrade. Adding it to this PR, so we don't have to deal with conflicts later.

**Pull request merged**